### PR TITLE
net/http contrib

### PR DESCRIPTION
**Pull request by gabsn**
`net/http` support for tracing thanks to @FreekingDean.

### CODE REVIEW

**Comment by FreekingDean** [comment]
Thank you gabsn for taking this on :)

**Comment by FreekingDean** [comment]
I have thought about this, and I wonder if we should build this into some form of a base handler for each of the router (httprouter/gorilla) packages. I know the main issue is that path routing would be messy as this does not take into account resource based paths: `/users/:id` in that this would cause a new resource for each user_id: `/users/1`, `/users/2`... I think we could alleviate that issue by allowing a path retrieval function passed into a private handler.

I'm not sure if I'm making sense so heres some code:

```go
//net/http/http.go
func (h *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
  ...
  tracedRequest, err := traceRequest(r, GetPath, h.tracer)
  ...
}

func GetPath(r *http.Request) string {
  return r.URL.Path
}

type PathFunc func(r *http.Request) string
func traceRequest(r *http.Request, pathFunc PathFunc, t *tracer.Tracer) (*http.Request, error) {
  ...
  span.SetMeta(ext.HTTPURL, pathFunc(r))
  ...
}
```
---
```go
//gorilla/muxtrace/muxtrace.go
import(
  ...
  httptrace "github.com/DataDog/dd-trace-go/libs/net/http"
)
func (m *MuxTracer) TraceHandleFunc(handler http.HandlerFunc) http.HandlerFunc {
  ...
  tracedRequest, err := httptrace.traceRequest(r, GetPath, m.tracer)
  ...
}

func GetPath(r *http.Request) string {
  route := mux.CurrentRoute(req)
  path, err := route.GetPathTemplate()
  if err != nil {
    // when route doesn't define a path
    path = "unknown"
  }
  return path
}
```

I hope this makes sense and I'm not just rambling & incoherent. 😅

**Comment by gabsn** [comment]
@FreekingDean Yes you're right, with this implementation we end up with a new resource per url hit.  I'll do a modification to be sure that `1 resource = 1 route`.

File: README.md [hunk 1]
```diff
@@ -1,16 +1,15 @@
...
+Sample code: https://godoc.org/github.com/DataDog/dd-trace-go/tracer#pkg-examples  
 List of integrations: https://godoc.org/github.com/DataDog/dd-trace-go/tracer#pkg-subdirectories
 
 ## Directory structure
 
-- [tracer](https://github.com/DataDog/dd-trace-go/tree/gabin/contributions/tracer): contains the low level API used to trace the different libraries.
+- [tracer](https://github.com/DataDog/dd-trace-go/tree/master/tracer): contains the low level API used to trace the different libraries.
 
-- [libs](https://github.com/DataDog/dd-trace-go/tree/gabin/contributions/libs): contains the different libraries supported by our APM solution. You just 
+- [libs](https://github.com/DataDog/dd-trace-go/tree/master/libs): contains the different libraries supported by our APM solution.
```
Comment (talwai) [inline README.md]: call this contrib

Comment (gabsn) [inline README.md]: I'll do that in another PR so it will be more clear. (it's a change of +100 files)

**Comment by palazzem** [comment]
@gabsn probably I'm missing something, but did we merge a broken build? I see that it ended with a failure: https://circleci.com/gh/DataDog/dd-trace-go/677

Indeed, the current master is broken: https://circleci.com/gh/DataDog/dd-trace-go/679

**Comment by gabsn** [comment]
Yeah the problem is not related with this PR, it's because of this [PR](https://github.com/grpc/grpc-go/pull/1392) on the grpc repo. We need to improve our testing to avoid those kind of issues, because for the moment circleci uses the `master` branch of the dependencies instead of a pinned version.

**Comment by palazzem** [comment]
Yea @gabsn, we should definitely improve our testing. I already have a plan about it and will write something down very soon so that we can address also other issues.

**Pull request merged**